pvlib/tests/test_irradiance.py

```diff
@@ -1235,6 +1235,12 @@ def test_clearsky_index():
     assert_series_equal(out, expected)
 
 
+def test_clearsky_index_renaming():
+    with pytest.warns(pvlibDeprecationWarning, match='ghi_clear'):
+        ghi, clearsky_ghi = 200, 300
+        irradiance.clearsky_index(ghi, clearsky_ghi)
```

Comment:
```suggestion
        irradiance.clearsky_index(ghi, clearsky_ghi=clearsky_ghi)
```
As positional arguments, it's all good to continue using them since the parameter hasn't changed position. But it's when using keyword arguments when things can go wrong. If a user had `clearsky_index(ghi=..., clearsky_ghi=...)`, now that the signature has changed, it would break and needs to be updated to the same, but using `ghi_clear` instead of the former should make it work.

The decorator is a shorthand to rename to the new param name if the old name was provided as the keyword param.

Reply:
ah of course that makes sense thanks @echedey-ls !!

Reply:
Thanks! I think part of the problem here was just my python knowledge and understanding of how to use decorators, in particular for deprecations. 
If you do develop something, perhaps you could use this PR as a base for the example once it's ready.
Now I know how to do this, I will complete #2274